src/core/server/services/queue/tasks/scraper/index.ts

```diff
@@ -0,0 +1,168 @@
+
+  logger.debug(
+    {
+      job_id: job.id,
+      job_name: JOB_NAME,
+      asset_id: id,
+      asset_url: url,
+      tenant_id: tenantID,
+    },
+    "starting to scrap the asset"
+  );
+
+  // Get the metadata from the scraped html.
+  const meta = await scraper.scrape(url);
+  if (!meta) {
+    logger.error(
+      {
+        job_id: job.id,
+        job_name: JOB_NAME,
+        asset_id: id,
+        asset_url: url,
+        tenant_id: tenantID,
+      },
+      "asset at specified url not found, can not scrape"
+    );
+    return;
+  }
+
+  // Update the Asset with the scraped details.
+  const asset = await updateAsset(options.mongo, tenantID, id, {
+    title: meta.title || undefined,
+    description: meta.description || undefined,
+    image: meta.image ? meta.image : undefined,
+    author: meta.author || undefined,
+    publication_date: meta.date ? new Date(meta.date) : undefined,
+    modified_date: meta.modified ? new Date(meta.modified) : undefined,
+    section: meta.section || undefined,
+    scraped: new Date(),
+  });
+  if (!asset) {
+    logger.error(
+      {
+        job_id: job.id,
+        job_name: JOB_NAME,
+        asset_id: id,
+        asset_url: url,
+        tenant_id: tenantID,
+      },
+      "asset at specified id not found, can not update with metadata"
+    );
+    return;
+  }
+
+  logger.debug(
+    {
+      job_id: job.id,
+      job_name: JOB_NAME,
+      asset_id: asset.id,
+      asset_url: url,
+      tenant_id: tenantID,
+    },
+    "scraped the asset"
+  );
+};
+
+export type Rule = Record<
+  string,
+  Array<
+    (options: { htmlDom: CheerioSelector; url: string }) => string | undefined
+  >
+>;
+
+class Scraper {
+  private rules: Rule[];
+
+  constructor(rules: Rule[]) {
+    this.rules = rules;
+  }
+
+  public async scrape(url: string) {
+    // Grab the page HTML.
+
+    // TODO: investigate adding scraping proxy support.
+    const res = await fetch(url, {});
+    if (res.status !== 200) {
+      return;
+    }
+
+    const html = await res.text();
+
+    // Load the DOM.
+    const htmlDom = cheerio.load(html);
+
+    // Gather the results by evaluating each of the rules.
+    const metadata: Record<string, string | undefined> = {};
+
+    for (const rule of this.rules) {
+      for (const property in rule) {
+        if (!rule.hasOwnProperty(property)) {
+          continue;
+        }
+
+        // Proceed through each of the properties and try to find the mapped
+        // properties.
+        for (const getter of rule[property]) {
+          const value = getter({ htmlDom, url });
+          if (value && value.length > 0) {
+            metadata[property] = value;
+
+            break;
+          }
+        }
+      }
+    }
+
+    return metadata;
+  }
+}
+
+export function createScraperTask(
+  queue: Queue.QueueOptions,
+  options: ScrapeProcessorOptions
+) {
+  // Create the scraper object.
+  const scraper = new Scraper([
+    require("metascraper-title")(),
```

Comment:
Not a full review, just encountered this while flying through. Why not stick to es6 modules?
```ts 
import metascraperTitle from "metascraper-title";

export function createScraperTask(
  queue: Queue.QueueOptions,
  options: ScrapeProcessorOptions
) {
  // Create the scraper object.
  const scraper = new Scraper([
    metascraperTitle(),
    ...
  ]);
```